src/runtime/runtime-debug.cc

```diff
@@ -1090,57 +1102,43 @@ RUNTIME_FUNCTION(Runtime_RecordReplayAssertExecutionProgress) {
     RecordReplayOnTargetProgressReached();
   }
 
-  if (!gRecordReplayAssertProgress) {
-    return ReadOnlyRoots(isolate).undefined_value();
-  }
-
-  if (!gProgressData) {
-    gProgressData = new std::vector<uint64_t>();
-  }
-  gProgressData->push_back(BuildScriptProgressEntry(args.at<JSFunction>(0)));
-
-  // FIXME the logic below should be gated on gRecordReplayCheckProgress when that exists.
-
-  HandleScope scope(isolate);
-  DCHECK_EQ(1, args.length());
-  Handle<JSFunction> function = args.at<JSFunction>(0);
-
-  Handle<SharedFunctionInfo> shared(function->shared(), isolate);
-  Handle<Script> script(Script::cast(shared->script()), isolate);
-  CHECK(RecordReplayHasRegisteredScript(*script));
-
-  Script::PositionInfo info;
-  Script::GetPositionInfo(script, shared->StartPosition(), &info,
-                          Script::WITH_OFFSET);
-  std::string name = GetScriptName(script);
-
-  if (!RecordReplayBytecodeAllowed()) {
-    recordreplay::Diagnostic("RecordReplayAssertExecutionProgress not allowed %s:%d:%d",
-                             name.c_str(), info.line + 1, info.column);
-  }
-  CHECK(RecordReplayBytecodeAllowed());
-
-  if (!gRecordReplayHasCheckpoint) {
-    recordreplay::Diagnostic("ExecutionProgress before first checkpoint %s:%d:%d",
-                             name.c_str(), info.line + 1, info.column);
-    CHECK(gRecordReplayHasCheckpoint);
-  }
+  if (gRecordReplayAssertProgress || gRecordReplayCheckProgress) {
+    CHECK_EQ(1, args.length());
```

Comment:
Let's remove the release mode check since this code is hot, this isn't ever going to catch anything.  I think it would also be worth removing a level of nesting and just having separate branches for gRecordReplayAssertProgress and gRecordReplayCheckProgress which both define the function local variable.  That is a tiny amount of code duplication but should be easier to follow and marginally faster.

Reply:
What is the "release mode check"?

Reply:
CHECK, CHECK_EQ etc. are release mode checks because they happen in all build configurations, including builds that are released to users.  Having a check like this use DCHECK_EQ instead would be fine, that's a debug mode check that only adds overhead in debug builds, and while we're not producing or recording/replaying using debug builds it's something we may explore in the future.